core/arch/arch.c

```diff
@@ -3355,7 +3355,8 @@ dr_mcontext_to_priv_mcontext(priv_mcontext_t *dst, dr_mcontext_t *src)
      * if we append to dr_mcontext_t in the future we'll need
      * to check src->size here.
      */
-    if (src->size != sizeof(dr_mcontext_t))
+    if (src->size != sizeof(dr_mcontext_t) &&
+        src->size != sizeof(dr_mcontext_t) - sizeof(dr_opmask_t))
```

Comment:
We failed to do the right thing in f9c8364fe41752979788a35d30163e59cb8e5b97 which widened the mcxt simd array fields.  We should support the prior size.  (In this PR seems fine to me.)

Reply:
Added size checks for above for each individual field and provided a compatibility copy for old case before f9c8364, copying dr_ymm_t size instead of dr_zmm_t.